src/ci/docker/host-x86_64/rfl/Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM ubuntu:23.10
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  g++ \
+  gcc-multilib \
+  make \
+  ninja-build \
+  file \
+  curl \
+  ca-certificates \
+  python3 \
+  git \
+  cmake \
+  gdb \
+  flex \
+  bison \
+  bc \
+  llvm-17 \
+  llvm-17-dev \
+  llvm-17-tools \
+  clang-17 \
+  lld-17 \
+  libelf-dev \
+  libedit-dev \
+  libssl-dev \
+  pkg-config \
+  zlib1g-dev \
+  xz-utils \
+  && rm -rf /var/lib/apt/lists/*
+
+COPY scripts/sccache.sh /scripts/
+RUN sh /scripts/sccache.sh
+
+# RfL needs access to llvm-ar
+ENV PATH="${PATH}:/usr/lib/llvm-17/bin"
+
+# We are disabling CI LLVM since this builder is intentionally using a host
+# LLVM, rather than the typical src/llvm-project LLVM.
```

Comment:
Is the choice of LLVM important / why are we using LLVM 17 here vs. 18 (current latest release?) And why use an external LLVM at all?

Reply:
I asked Jakub about this part when we working on this; and he said it was taken from the original job, but that it would be useful to avoid building our own LLVM. I don't think the version matters much from the kernel point of view, especially for the few things we are actually doing here.

Reply:
If this isn't part of PR CI (which I think makes sense, hopefully we're not breaking RfL *that* often that we want the early warning), then building or not building LLVM shouldn't really matter since that'll be cached anyway.

Reply:
Currently the build takes about 12 minutes, even a fully (sccache-)cached LLVM build will take similar time (at least 10 minutes including linking, IIRC), so it would double the build duration. But if you don't mind that, I can switch it to locally built LLVM. Or we can just use `download-ci-llvm`.

Reply:
Test builders are usually using download-ci-llvm today, so I think we'd get the faster build?

Reply:
Yes, that should work, thanks.